io-engine/src/core/snapshot.rs

```diff
@@ -64,23 +99,63 @@ pub trait SnapshotOps {
     fn prepare_snap_config(
         &self,
         snap_name: &str,
+        entity_id: &str,
         txn_id: &str,
     ) -> Option<SnapshotParams>;
+
+    /// List Snapshots.
+    fn list_snapshot(self) -> Vec<VolumeSnapshotDescriptor>;
```

Comment:
Why do we consume Lvol object when listing its snapshots ?

Reply:
> Why do we consume Lvol object when listing its snapshots ?

To filter lvol whose parent is input lvol to list snapshot specific to current lvol.

Reply:
'create_snapshot()' (and most of the other methods) uses '&self', but here we use 'self', i.e. the object is consumed. This is not fully aligned with Lvol API, IMHO.

Reply:
Accepted. Changed the signature.